pkg/signup/signup_service.go

```diff
@@ -238,6 +242,45 @@ func (s *ServiceImpl) GetSignup(userID string) (*Signup, error) {
 	return signupResponse, nil
 }
 
+// PostVerification returns userSignup resource from the host cluster.
+// Returns the usersignup, an http code and an error if present.
+func (s *ServiceImpl) PostVerification(dailyLimit int, responseBody map[string]string, userID, code string) (*crtapi.UserSignup, int, error) {
+	// Retrieve UserSignup resource from the host cluster
+	userSignup, err := s.UserSignups.Get(userID)
+	if err != nil {
+		return nil, http.StatusNotFound, err
+	}
+
+	// get the annotation counter
+	annotationCounter := userSignup.Annotations[crtapi.UserSignupVerificationCounterAnnotationKey]
+	counter := 0
+	if annotationCounter != "" {
+		counter, err = strconv.Atoi(annotationCounter)
+		if err != nil {
+			return nil, http.StatusInternalServerError, err
+		}
+	}
+
+	// check if counter has exceeded the limit of daily limit - if at limit error out
+	if counter > dailyLimit {
+		return nil, http.StatusForbidden, errors2.New("daily limit has been exceeded")
+	}
+
+	// set the usersignup annotations
+	userSignup.Annotations[crtapi.UserSignupVerificationCounterAnnotationKey] = strconv.Itoa(counter + 1)
+	userSignup.Annotations[crtapi.UserSignupPhoneNumberLabelKey] = responseBody["country_code"] + responseBody["phone_number"]
+	userSignup.Annotations[crtapi.UserSignupVerificationCodeAnnotationKey] = code
+	userSignup.Annotations[crtapi.UserSignupVerificationTimestampAnnotationKey] = time.Now().Format(verification.TimestampLayout)
```

Comment:
All of these verification-related annotations should be managed by the verification service.

Reply:
fixed here: https://github.com/codeready-toolchain/registration-service/blob/60ca7377aabe427ed8f8e3d642380b1b8396454e/pkg/controller/signup.go#L51